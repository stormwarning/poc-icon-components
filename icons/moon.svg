<svg width="32" height="32" viewBox="0 0 32 32" fill="none" xmlns="http://www.w3.org/2000/svg">
<g id="Container=None, Appearance=Outlined">
<g id="symbol">
<mask id="path-1-inside-1_1306_32901" fill="white">
<path d="M25.5828 19.7723C25.759 19.3552 25.2944 18.9782 24.8778 19.1557C23.7749 19.6254 22.5611 19.8853 21.2866 19.8853C16.2211 19.8853 12.1147 15.7789 12.1147 10.7134C12.1147 9.43889 12.3746 8.22513 12.8444 7.12221C13.0218 6.70563 12.6448 6.24097 12.2278 6.4172C8.56816 7.96346 6 11.5863 6 15.8089C6 21.4373 10.5627 26 16.1911 26C20.4137 26 24.0365 23.4318 25.5828 19.7723Z"/>
</mask>
<path d="M24.8778 19.1557L25.6615 20.9957L24.8778 19.1557ZM25.5828 19.7723L23.7405 18.9938L25.5828 19.7723ZM24.0941 17.3156C23.2344 17.6817 22.2869 17.8853 21.2866 17.8853V21.8853C22.8353 21.8853 24.3153 21.5691 25.6615 20.9957L24.0941 17.3156ZM21.2866 17.8853C17.3257 17.8853 14.1147 14.6743 14.1147 10.7134H10.1147C10.1147 16.8835 15.1165 21.8853 21.2866 21.8853V17.8853ZM14.1147 10.7134C14.1147 9.71311 14.3183 8.76557 14.6844 7.90592L11.0043 6.3385C10.4309 7.6847 10.1147 9.16468 10.1147 10.7134H14.1147ZM8 15.8089C8 12.4181 10.0608 9.50399 13.0062 8.2595L11.4493 4.5749C7.07552 6.42294 4 10.7545 4 15.8089H8ZM16.1911 24C11.6673 24 8 20.3327 8 15.8089H4C4 22.5419 9.45814 28 16.1911 28V24ZM23.7405 18.9938C22.496 21.9392 19.5819 24 16.1911 24V28C21.2456 28 25.5771 24.9245 27.4251 20.5507L23.7405 18.9938ZM14.6844 7.90592C15.0913 6.95056 14.8613 5.9331 14.2833 5.25263C13.6748 4.53617 12.5725 4.10032 11.4493 4.5749L13.0062 8.2595C12.3001 8.55785 11.5949 8.26622 11.2347 7.84218C10.9051 7.45412 10.7748 6.87728 11.0043 6.3385L14.6844 7.90592ZM25.6615 20.9957C25.1227 21.2252 24.5459 21.0949 24.1578 20.7653C23.7338 20.4051 23.4421 19.7 23.7405 18.9938L27.4251 20.5507C27.8997 19.4275 27.4638 18.3253 26.7474 17.7167C26.0669 17.1387 25.0494 16.9087 24.0941 17.3156L25.6615 20.9957Z" fill="black" mask="url(#path-1-inside-1_1306_32901)"/>
</g>
</g>
</svg>
