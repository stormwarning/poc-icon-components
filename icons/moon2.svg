<svg width="32" height="32" viewBox="0 0 32 32" fill="none" xmlns="http://www.w3.org/2000/svg">
<path fill-rule="evenodd" clip-rule="evenodd" d="M24.8778 19.1557C25.2944 18.9782 25.759 19.3552 25.5828 19.7723C24.0365 23.4318 20.4137 26 16.1911 26C10.5627 26 6 21.4373 6 15.8089C6 11.5863 8.56816 7.96346 12.2278 6.4172C12.6448 6.24097 13.0218 6.70563 12.8444 7.12221C12.3746 8.22513 12.1147 9.43889 12.1147 10.7134C12.1147 15.7789 16.2211 19.8853 21.2866 19.8853C22.5611 19.8853 23.7749 19.6254 24.8778 19.1557ZM21.2866 21.8878C21.4224 21.8878 21.5578 21.8854 21.6925 21.8805C20.2377 23.1999 18.3072 24.0025 16.1911 24.0025C11.6673 24.0025 8 20.3352 8 15.8114C8 13.6953 8.80261 11.7648 10.122 10.31C10.1171 10.4447 10.1147 10.5801 10.1147 10.7159C10.1147 16.886 15.1165 21.8878 21.2866 21.8878Z" fill="black"/>
</svg>
