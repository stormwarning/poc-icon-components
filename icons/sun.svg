<svg width="32" height="32" viewBox="0 0 32 32" fill="none" xmlns="http://www.w3.org/2000/svg">
<g id="Container=None, Appearance=Outlined">
<path id="symbol" fill-rule="evenodd" clip-rule="evenodd" d="M16 4.5C16.5523 4.5 17 4.94772 17 5.5V8.06189C16.6724 8.02104 16.3387 8 16 8C15.6613 8 15.3276 8.02104 15 8.06189V5.5C15 4.94772 15.4477 4.5 16 4.5ZM20.9055 9.68004C21.4337 10.0906 21.9091 10.566 22.3198 11.0942L24.1315 9.28248C24.522 8.89196 24.522 8.2588 24.1315 7.86827C23.741 7.47775 23.1078 7.47775 22.7173 7.86827L20.9055 9.68004ZM24 16C24 16.3387 23.979 16.6724 23.9381 17H26.5C27.0523 17 27.5 16.5523 27.5 16C27.5 15.4477 27.0523 15 26.5 15H23.9381C23.979 15.3276 24 15.6613 24 16ZM22.3198 20.9057C21.9092 21.4339 21.4338 21.9093 20.9056 22.3199L22.7174 24.1317C23.1079 24.5223 23.7411 24.5223 24.1316 24.1317C24.5221 23.7412 24.5221 23.108 24.1316 22.7175L22.3198 20.9057ZM16 24C15.6613 24 15.3276 23.979 15 23.9381V26.5C15 27.0523 15.4477 27.5 16 27.5C16.5523 27.5 17 27.0523 17 26.5V23.9381C16.6724 23.979 16.3387 24 16 24ZM11.0942 22.3198C10.566 21.9091 10.0906 21.4337 9.68004 20.9055L7.86804 22.7175C7.47752 23.108 7.47752 23.7412 7.86804 24.1317C8.25857 24.5223 8.89173 24.5223 9.28226 24.1317L11.0942 22.3198ZM8 16C8 15.6613 8.02104 15.3276 8.06189 15H5.5C4.94772 15 4.5 15.4477 4.5 16C4.5 16.5523 4.94772 17 5.5 17H8.06189C8.02104 16.6724 8 16.3387 8 16ZM9.68009 11.0944C10.0907 10.5662 10.5661 10.0908 11.0943 9.68019L9.28238 7.86827C8.89185 7.47775 8.25869 7.47775 7.86816 7.86827C7.47764 8.2588 7.47764 8.89196 7.86816 9.28249L9.68009 11.0944ZM20 16C20 18.2091 18.2091 20 16 20C13.7909 20 12 18.2091 12 16C12 13.7909 13.7909 12 16 12C18.2091 12 20 13.7909 20 16ZM22 16C22 19.3137 19.3137 22 16 22C12.6863 22 10 19.3137 10 16C10 12.6863 12.6863 10 16 10C19.3137 10 22 12.6863 22 16Z" fill="black"/>
</g>
</svg>
