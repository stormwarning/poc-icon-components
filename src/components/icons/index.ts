export { IconMoon } from './IconMoon/IconMoon';
export { IconMoon2 } from './IconMoon2/IconMoon2';
export { IconSun } from './IconSun/IconSun';
