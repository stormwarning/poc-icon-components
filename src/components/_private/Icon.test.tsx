import React from 'react';

import { render, screen } from '../../../utils/test-utils';
import * as Icons from '../icons';

describe('<Icon />', () => {
  it('renders at the correct size', () => {
    const { rerender } = render(<Icons.IconMoon data-testid="moon" />);
    const icon = screen.getByTestId('moon');

    expect(icon).toBeInstanceOf(SVGSVGElement);
    expect(icon).toHaveAttribute('width', '32');

    rerender(<Icons.IconMoon size="16" data-testid="moon" />);
    expect(icon).toHaveAttribute('width', '16');
  });

  it('hides icons from assistive tech when a title is not provided', () => {
    render(<Icons.IconMoon data-testid="moon" />);

    expect(screen.getByTestId('moon')).toHaveAttribute('aria-hidden');
  });

  it('applies an ARIA role attribute when passed a label', () => {
    render(<Icons.IconMoon label="Dark" labelId="theme" data-testid="moon" />);

    const icon = screen.getByTestId('moon');

    expect(icon).toHaveAttribute('role', 'img');
    expect(icon).toHaveAttribute('aria-labelledby', 'theme');
    expect(screen.getByText('Dark')).toHaveAttribute('id', 'theme');
  });
});

Object.keys(Icons).forEach((icon) => {
  const IconComponent = Icons[icon];

  // eslint-disable-next-line jest/valid-title
  describe(`<${IconComponent.name} />`, () => {
    it('should match snapshot', async () => {
      expect.assertions(1);
      const { container } = render(<IconComponent />);
      expect(container.firstElementChild).toMatchSnapshot();
    });
  });
});
