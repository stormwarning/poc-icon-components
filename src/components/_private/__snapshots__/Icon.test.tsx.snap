// Vitest Snapshot v1

exports[`<IconMoon /> > should match snapshot 1`] = `
<svg
  aria-hidden="true"
  height="32"
  width="32"
>
  <use
    xlink:href="/icon-sprite.svg#moon"
  />
</svg>
`;

exports[`<IconMoon2 /> > should match snapshot 1`] = `
<svg
  aria-hidden="true"
  height="32"
  width="32"
>
  <use
    xlink:href="/icon-sprite.svg#moon2"
  />
</svg>
`;

exports[`<IconSun /> > should match snapshot 1`] = `
<svg
  aria-hidden="true"
  height="32"
  width="32"
>
  <use
    xlink:href="/icon-sprite.svg#sun"
  />
</svg>
`;
